crates/bevy_ecs/src/change_detection.rs

```diff
@@ -0,0 +1,152 @@
+use crate::component::{Component, ComponentTicks};
+use bevy_reflect::Reflect;
+use std::ops::{Deref, DerefMut};
+
+pub trait ChangeDetectable {
```

Comment:
I would like to see some brief doc comments explaining what this trait does and how an end user (or engine dev) can apply it.

Reply:
(just updated it and would love feedback on my explanation).
I'm not the best with words tbh.... 😆